server/src/main/java/org/elasticsearch/index/mapper/RoutingFields.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.IndexSettings;
+
+import java.net.InetAddress;
+
+/**
+ * Collects fields contributing to routing from documents.
+ */
+public interface RoutingFields {
```

Comment:
This interface looks almost the same to `DocumentDimensions`.
Maybe do a git rename instead of removing and adding a few file? That way file history is maintained.

Reply:
I wish IntelliJ did that automatically when copying a file.. 

Tried to delete and `git mv`, I still get this.. Git preserves current history for this file.

Reply:
Strange. A rename shouldn't show up as a removal of old file and addition of a new file. 
But this isn't a real issue.